sdk/identity/Azure.Identity/src/SharedTokenCacheCredential.cs

```diff
@@ -133,7 +133,7 @@ private async ValueTask<ExtendedAccessToken> GetTokenImplAsync(TokenRequestConte
             }
             catch (MsalUiRequiredException)
             {
-                return new ExtendedAccessToken(scope.Failed(new CredentialUnavailableException($"Token acquisition failed for user {_username}. To fix, re-authenticate through developer tooling supporting Azure single sign on.")));
+                return new ExtendedAccessToken(scope.Failed(new CredentialUnavailableException($"{nameof(SharedTokenCacheCredential)} authentication unavailable, token acquisition failed for user {_username}. To fix, re-authenticate through developer tooling supporting Azure single sign on.")));
```

Comment:
What do we get if username is null?

Reply:
If we get this exception it means we already have picked an account, and so have already set `_username`.